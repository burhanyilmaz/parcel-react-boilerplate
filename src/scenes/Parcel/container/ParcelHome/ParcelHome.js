import React, { Component } from 'react';
import '../../style/style.css';

class ParcelHome extends Component {

    increaseCounter() {
        const { counter, changeCounter } = this.props;
        return changeCounter(counter + 1);
    }

    decreaseCounter() {
        const { counter, changeCounter } = this.props;
        return changeCounter(counter - 1);
    }

    render() {
        const { counter } = this.props;

        return(
            <div>
               <h3>Parcel React Boilerplate</h3>
                <div className="container">
                    <div className="column minus" onClick={() => this.decreaseCounter()}>
                        <span>âˆ’</span>
                    </div>
                    <div className="column">
                        <h1 className="counter">{counter}</h1>
                    </div> 
                    <div className="column plus" onClick={() => this.increaseCounter()}>
                        <span>+</span>                    
                    </div>
               </div>
            </div>
        );
    }
}

export default ParcelHome;