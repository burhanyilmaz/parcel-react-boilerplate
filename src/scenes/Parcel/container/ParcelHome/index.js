import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { STATE_PARCEL } from '../../constants';
import { changeCounter } from '../../actions/ParcelActions';
import ParcelHome from './ParcelHome';

const mapStateToProps = state => {
    const { counter } = state[STATE_PARCEL];

    return { counter };
}

const mapDispatchToProps = (dispatch) => {
    return {
        changeCounter: bindActionCreators(changeCounter, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(ParcelHome);
