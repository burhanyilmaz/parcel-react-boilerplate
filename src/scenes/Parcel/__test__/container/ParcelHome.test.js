import React from 'react';
import { shallow } from 'enzyme';
import ParcelHome from '../../container/ParcelHome/ParcelHome';


describe("ParcelHone Component Testing", () => {
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<ParcelHome />)
        
    })

    it ('ParcelHome component snapshot testing', function() {
        expect(wrapper).toMatchSnapshot();
    });

    it ("ParcelHome'da bulunan string'lerin kontrolü.", () => {
        const wrapper = shallow(<ParcelHome />);

        expect(wrapper.text()).toContain("Parcel React Boilerplate");
        expect(wrapper.text()).toContain("+");
        expect(wrapper.text()).toContain("−");
    });
});
