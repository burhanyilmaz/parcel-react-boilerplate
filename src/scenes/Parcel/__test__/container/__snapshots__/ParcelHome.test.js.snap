// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParcelHone Component Testing ParcelHome component snapshot testing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParcelHome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Parcel React Boilerplate
        </h3>,
        <div
          className="container"
        >
          <div
            className="column minus"
            onClick={[Function]}
          >
            <span>
              −
            </span>
          </div>
          <div
            className="column"
          >
            <h1
              className="counter"
            />
          </div>
          <div
            className="column plus"
            onClick={[Function]}
          >
            <span>
              +
            </span>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Parcel React Boilerplate",
        },
        "ref": null,
        "rendered": "Parcel React Boilerplate",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="column minus"
              onClick={[Function]}
            >
              <span>
                −
              </span>
            </div>,
            <div
              className="column"
            >
              <h1
                className="counter"
              />
            </div>,
            <div
              className="column plus"
              onClick={[Function]}
            >
              <span>
                +
              </span>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                −
              </span>,
              "className": "column minus",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "−",
              },
              "ref": null,
              "rendered": "−",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="counter"
              />,
              "className": "column",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "counter",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                +
              </span>,
              "className": "column plus",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
              },
              "ref": null,
              "rendered": "+",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Parcel React Boilerplate
          </h3>,
          <div
            className="container"
          >
            <div
              className="column minus"
              onClick={[Function]}
            >
              <span>
                −
              </span>
            </div>
            <div
              className="column"
            >
              <h1
                className="counter"
              />
            </div>
            <div
              className="column plus"
              onClick={[Function]}
            >
              <span>
                +
              </span>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Parcel React Boilerplate",
          },
          "ref": null,
          "rendered": "Parcel React Boilerplate",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="column minus"
                onClick={[Function]}
              >
                <span>
                  −
                </span>
              </div>,
              <div
                className="column"
              >
                <h1
                  className="counter"
                />
              </div>,
              <div
                className="column plus"
                onClick={[Function]}
              >
                <span>
                  +
                </span>
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  −
                </span>,
                "className": "column minus",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "−",
                },
                "ref": null,
                "rendered": "−",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="counter"
                />,
                "className": "column",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "counter",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  +
                </span>,
                "className": "column plus",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                },
                "ref": null,
                "rendered": "+",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
